af-plans/juiceshop-selenium-auth/README.md

```diff
@@ -0,0 +1,85 @@
+# Juice Shop Selenium Authentication
+
+This directory contains [Automation Framework](https://www.zaproxy.org/docs/automate/automation-framework/) yaml plans and a set of scripts to demonstrate how to set up ZAP to automate authentication using Selenium to launch browsers.
+
+Note that ZAP _can_ authenticate to Juice Shop _without_ using Selenium, and normally this is what we would recommend.
+However in this case we are using it to show how to use Selenium to authenticate to your app if there is no alternative.
+
+## Requirements
+
+To run this plan you need to:
+
+* Install Java 11 - unfortunately we currently have to use Oracle Nashorn for the Selenium scripting which is not available in later versions of Java
+* Install ZAP
+* Run [Juice Shop](https://github.com/juice-shop/juice-shop) so that iit is accessible via http://localhost:3000/
+* Add a user to Juice Shop with the credentials:
+  * username: test@test.com
+  * password: test123
+* Edit all of the yaml files and shell scripts to replace `/full/path/` to be the local full path of your scripts
+* Edit all of the shell scripts to correct the path to your ZAP installation
+* Ensure that the user credentials are available via the environmental variables (these _should_ be set up correctly via the supplied scripts):
+  * JS_USER: test@test.com
+  * JS_PWD: test123
+
+## Test Run
+
+With the above requirements in place run the script `./js-test.js` (Linux or MacOS) or `js-test.bat` (Windows).
+
+This uses the `juiceshop-test-req.yaml` file which sets up the environment and just makes one authenticated request.
+
+A set of [Job Tests](https://www.zaproxy.org/docs/desktop/addons/automation-framework/tests/) check that the authenticate worked as expected.
+
+You should see messages on the command line like:
+
+```
+Job requestor test of type stats passed: At least 1 successful login [1 >= 1]
+Job requestor test of type stats passed: No login failures [0 <= 0]
+Job requestor finished
+Automation plan succeeded!
+```
+
+These indicate that the plan ran successfully.
+
+Note that Firefox reports lots of messages to the command line - these are difficult to suppress :/
+
+To show that ZAP fails the plan if the authentication fails, either do not register the user or change the script to use different credentials.
+
+## Full Run
+
+With the above requirements in place run the script `./js-auth.js` (Linux or MacOS) or `js-auth.bat` (Windows).
+
+This uses the `juiceshop-auth.yaml` file which sets up the environment and runs both the standard and ajax spider,
+both of which authenticate using the `test@test.com` user.
+
+A set of [Job Tests](https://www.zaproxy.org/docs/desktop/addons/automation-framework/tests/) check that the authenticate worked as expected for both spiders.
+
+You should see messages on the command line like:
```

Comment:
```suggestion
You should see messages in the console like:
```

Reply:
Couldnt that be read as the ZAP script console? In this case these messages will be writen to stdout..

Reply:
I'm all for getting it right 😁

Reply:
@thc202 has suggested "standard output" as it depends how ZAP is launched - that works for me..

Reply:
Okay, lets go that way